### ISSUE TITLE
oceSetData() error on length of names vector

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1110]
The code
```R
library(ctd)
data("ctd")
d <- oceSetData(ctd, name="fake", value=seq_along(ctd[["salinity"]]),
                unit=list(unit=expression(), scale=""),
                originalName="FAKE")
```
produces the error
```R
Error in names(object@metadata$dataNamesOriginal) <- names(object@data) : 
  'names' attribute [8] must be the same length as the vector [7]
```
which I intend to fix today.

Note to CR: I found this because of the (new) code-coverage tests ... I am on a mission to add tests that will exercise the central elements of oce code. For example, https://codecov.io/gh/dankelley/oce/src/6c6815451c94b9d9458c87469a3ca76c2d7e4b44/R/accessors.R shows that `accessors.R` was not being exercised very much, so I added a test for `oceSetData` and, voila -- an error!  (NOTE: I have not yet got local code-coverage tests working--see e.g. https://github.com/jimhester/covr/issues/231#issuecomment-258635816 -- but once I figure that out, this is going to be a very useful addition to our development methodology).

**Comment by dankelley** [comment]
PS I started a [wiki item](https://github.com/dankelley/oce/wiki/code-coverage) on trying to get code coverage to work.

**Comment by dankelley** [comment]
Fixed in `develop` commit a880e7a9d526dfd5aee9bdc2480211bb8e2e6fb8
